src/DependencyInjection/SonataHelpersExtension.php

```diff
@@ -45,11 +45,13 @@ public function prepend(ContainerBuilder $container)
         $config            = $container->getExtensionConfig($this->getAlias())[0];
         $sonataMediaConfig = $container->getExtensionConfig('sonata_media')[0];
 
-        if (empty($config['sonata_media_private_file_provider']['allowed_extensions'])) {
-            $config['sonata_media_private_file_provider']['allowed_extensions'] = $sonataMediaConfig['providers']['file']['allowed_extensions'];
-        }
-        if (empty($config['sonata_media_private_file_provider']['allowed_mime_types'])) {
-            $config['sonata_media_private_file_provider']['allowed_mime_types'] = $sonataMediaConfig['providers']['file']['allowed_mime_types'];
+        if (\array_key_exists('providers', $sonataMediaConfig)) {
```

Comment:
i suppose that existence of `file` key should be tested too  as it may not exist too

Reply:
maybe 

`$sonataMediaConfig['providers']['file']['allowed_extensions'] ?? [];`

is a good solution no? WDYT?

Reply:
That would set it to an empty array (instead of not setting it at all).

I suggest:
```
foreach(['allowed_extensions', 'allowed_mime_types'] as $key) {
    if(empty($sonataMediaConfig['providers']['file'][$key])
    || !empty($config['sonata_media_private_file_provider'][$key])) {
        continue;
    }

    $config['sonata_media_private_file_provider'][$key] = $sonataMediaConfig['providers']['file'][$key];
}
```

Reply:
it's better to set it to an empty array as it would be consistent with default values exposed in Configuration class. (and would be the opposite of cannotBeEmpty you added)